src/Resolvers/Collections.ts

```diff
@@ -75,6 +75,36 @@ export class CollectionsResolver {
     return await this.repository.findOne({ slug })
   }
 
+  @FieldResolver(type => [Collection])
+  async relatedCollections(
+    @Root() collection: Collection
+  ): Promise<Collection[]> {
+    const isArtistBased =
+      collection.query.artist_ids && collection.query.artist_ids.length > 0
+
+    if (isArtistBased) {
+      const artistResults = await this.repository.find({
+        where: { "query.artist_ids": { $in: collection.query.artist_ids } },
+      })
+
+      const relatedArtistCollections = reject(artistResults, {
+        id: collection.id,
+      })
+
+      if (relatedArtistCollections.length > 4) {
+        return relatedArtistCollections
+      }
+    }
```

Comment:
I think you should have a fallback return here. like `return []` otherwise if there are less than 4 related artist collections, it'll continue and fetch related category collections

Reply:
That's what we would want based on the requirements, right? If the collection is associated with an artist and that artist has at least 5 collections display those otherwise fallback to display collections with the same category [GROW-1228](https://artsyproduct.atlassian.net/browse/GROW-1228).

Reply:
Yup, you're right. looks good then